### PR TITLE
Add pyslalib

### PR DESCRIPTION
**Pull request by ickc**
@conda-forge/help-python

c.f. #3812

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/pyslalib/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('fortran') }}
+    - numpy
```
Comment (chrisburr) [inline recipes/pyslalib/meta.yaml:21]: I'm not sure what the problem is but this definitely shouldn't be in build
```suggestion
```

File: recipes/pyslalib/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    - {{ pin_compatible('numpy') }}
+
+test:
+  imports:
+    - pyslalib
+  commands:
+    - pip check
+  requires:
+    - pip
+    - {{ pin_compatible('numpy') }}
```
Comment (chrisburr) [inline recipes/pyslalib/meta.yaml:37]: Similarly:
```suggestion
```

File: recipes/pyslalib/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/pySLALIB-{{ version }}.tar.gz
+  sha256: 21d0e1ae6fc73ade9c90ec9e956b2da33163383388cc84e5ec58994bf66c8b35
+
+build:
+  number: 0
```
Comment (chrisburr) [inline recipes/pyslalib/meta.yaml:14]: It looks like the upstream setup.py just doesn't support Windows at all causing the build to not actually compile any of the sources. If you care you'll need to patch it else I'd suggest just skipping it:
```suggestion
  skip: true  # [win]
  number: 0
```

**Comment by ickc** [comment]
@chrisburr, thanks! Ready to merge now.

**Pull request merged**